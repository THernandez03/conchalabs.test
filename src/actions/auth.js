import { SIGN_IN, SIGN_OUT } from '../constants/actionTypes';

export const signIn = (payload) => ({ type: SIGN_IN, payload });
export const signOut = (payload) => ({ type: SIGN_OUT, payload });
