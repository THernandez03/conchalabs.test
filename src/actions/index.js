export * from './musicControls';
export * from './auth';
export * from './ui';
