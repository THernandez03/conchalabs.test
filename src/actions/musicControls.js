import {
  TOGGLE_MUSIC,
  PLAY_MUSIC,
  PAUSE_MUSIC,
} from '../constants/actionTypes';

export const toggle = () => ({ type: TOGGLE_MUSIC });
export const play = () => ({ type: PLAY_MUSIC });
export const pause = () => ({ type: PAUSE_MUSIC });
