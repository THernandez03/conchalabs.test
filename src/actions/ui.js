import {
  SIDEBAR_TOGGLE,
  SIDEBAR_OPEN,
  SIDEBAR_CLOSE,
  SELECT_ITEM,
  DESELECT_ITEM,
} from '../constants/actionTypes';

export const toggleSidebar = (payload) => ({ type: SIDEBAR_TOGGLE, payload });
export const showSidebar = (payload) => ({ type: SIDEBAR_OPEN, payload });
export const hideSidebar = (payload) => ({ type: SIDEBAR_CLOSE, payload });

export const selectItem = (payload) => ({ type: SELECT_ITEM, payload });
export const deselectItem = (payload) => ({ type: DESELECT_ITEM, payload });
