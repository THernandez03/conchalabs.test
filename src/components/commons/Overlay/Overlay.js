import { useDispatch } from 'react-redux';
import styled from '@emotion/styled';
import PropTypes from 'prop-types';

import { hideSidebar } from '../../../actions';

export const Overlay = styled(({ className }) => {
  const dispatch = useDispatch();

  const handleOverlaySelected = () => {
    dispatch(hideSidebar());
  };

  return (
    <div
      className={className}
      onClick={handleOverlaySelected}
      onKeyDown={handleOverlaySelected}
    />
  );
})`
  background: rgba(0, 0, 0, 0.8);
  width: 100%;
  height: 100%;
  position: absolute;
  top: 0;
  left: 0;
  z-index: 1;
`;

Overlay.defaultProps = {
  className: PropTypes.string,
};

Overlay.defaultProps = {};
