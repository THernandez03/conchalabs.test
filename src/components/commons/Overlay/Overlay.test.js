import { render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import { Overlay } from './Overlay';
import { hideSidebar } from '../../../actions';

jest.mock('react-redux', () => ({ useDispatch: () => jest.fn() }));
jest.mock('../../../actions', () => ({ hideSidebar: jest.fn() }));

describe('<Overlay/>', () => {
  it('Should render the componente', () => {
    const { container } = render(<Overlay />);
    expect(container).toMatchSnapshot();
  });

  it('Should dispatch hideSidebar when clicked over the component', () => {
    const { container } = render(<Overlay />);
    userEvent.click(container.firstChild);
    expect(hideSidebar).toHaveBeenCalled();
  });
});
