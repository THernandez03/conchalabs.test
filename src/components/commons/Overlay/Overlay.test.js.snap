// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Overlay/> Should render the componente 1`] = `
<div>
  <div
    class="css-27c4mj-Overlay e1otb7v0"
  />
</div>
`;
